daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -367,35 +367,52 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
     } -- localContracts.keySet
 
   // This method visits to all nodes of the transaction in execution order.
-  // Exercise nodes are visited twice: when execution reaches them and when execution leaves their body.
+  // Exercise/rollback nodes are visited twice: when execution reaches them and when execution leaves their body.
+  // On the first visit of an execution/rollback node, the caller can prevent traversal of the children
   final def foreachInExecutionOrder(
-      exerciseBegin: (Nid, Node.NodeExercises[Nid, Cid]) => Unit,
+      exerciseBegin: (Nid, Node.NodeExercises[Nid, Cid]) => Boolean,
+      rollbackBegin: (Nid, Node.NodeRollback[Nid]) => Boolean,
       leaf: (Nid, Node.LeafOnlyNode[Cid]) => Unit,
       exerciseEnd: (Nid, Node.NodeExercises[Nid, Cid]) => Unit,
+      rollbackEnd: (Nid, Node.NodeRollback[Nid]) => Unit,
```

Comment:
@andreaslochbihler-da @oliverse-da are you fine with changing the signature of this method or do you prefer we create a separate method ?

Reply:
Should be fine and I'm always in favour of keeping interfaces small. If needed, we can implement the old signature using the new one.